android/src/main/java/com/google/android/react/navsdk/NavViewManager.java

```diff
@@ -310,19 +327,31 @@ public Map<String, Object> getExportedCustomDirectEventTypeConstants() {
   }
 
   /** Replace your React Native view with a custom fragment */
-  public void createFragment(FrameLayout root, Map stylingOptions) {
+  public void createFragment(FrameLayout root, Map stylingOptions, int fragmentType) {
     setupLayout(root);
 
     FragmentActivity activity = (FragmentActivity) reactContext.getCurrentActivity();
     if (activity != null) {
       int viewId = root.getId();
-      NavViewFragment fragment = new NavViewFragment(reactContext, root.getId());
-      fragmentMap.put(viewId, new WeakReference<NavViewFragment>(fragment));
+      Fragment fragment;
+      // FragmentType 0 = MAP, 1 = NAVIGATION.
+      if (fragmentType == 0) {
```

Comment:
Nit: Please add a constant (or enum) we can reference here.

Reply:
Yeah I think it's better that way. I decided to leave it out as it seemed there were no custom types used before in the native code side. Anyway I'll change this.

Reply:
Done.